hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterContext.java

```diff
@@ -412,6 +422,19 @@ private void cancelExecute(long jobId, long executionId) {
         });
     }
 
+    /**
+     * Cancels the execution invocation in order to restart it afterwards if the job is currently being executed
+     */
+    boolean cancelCurrentExecution() {
+        ExecutionCallback<Object> callback = this.executionCancellationCallback;
+        if (callback != null) {
+            callback.onFailure(null);
```

Comment:
we should add a comment here. Do you think having an explict exception would be better? I.e. JobRestartRequestedException ? Also naming is still a bit ambiguous between `cancelExecute` and cancelCurrentExecution`.